src/app/loadout/ModPicker.tsx

```diff
@@ -82,23 +82,26 @@ function mapStateToProps() {
       plugCategoryHashWhitelist,
       currentStore
     ): PlugSet[] => {
-      const artificeString = defs?.InventoryItem.get(3727270518).displayProperties.name;
-
       const plugSets: { [plugSetHash: number]: PlugSet } = {};
       if (!profileResponse || !defs) {
-        return [];
+        return emptyArray();
       }
 
+      // We need the name of the Artifice armor perk to show in one of the headers
+      const artificeString = defs?.InventoryItem.get(3727270518)?.displayProperties.name;
+
+      // Look at every armor item and see what sockets it has
       for (const item of allItems) {
         if (
-          !item ||
-          !item.sockets ||
+          !item?.sockets ||
           // Makes sure it's an armour 2.0 item
           !isLoadoutBuilderItem(item) ||
           // If classType is passed in, only use items from said class,
           // otherwise use items from all characters.
           // Useful if in loadouts and only mods and guns
-          !(classType === DestinyClass.Unknown || item.classType === classType)
+          (classType !== DestinyClass.Unknown &&
+            classType !== undefined &&
+            item.classType !== classType)
```

Comment:
This said "if classType is passed in" but if classType was undefined (not passed in) it'd filter out everything